src/main/java/org/opentripplanner/api/resource/GraphPathToTripPlanConverter.java

```diff
@@ -329,6 +330,18 @@ public static CoordinateArrayListSequence makeCoordinates(Edge[] edges) {
                 if (backMode != TraverseMode.LEG_SWITCH) {              // Start of leg switch
                     legIndexPairs[1] = i;
                 } else if (forwardMode != TraverseMode.LEG_SWITCH) {    // End of leg switch
+                    // if the start edge of the first state in the current leg index pair is a TransitBoardAlight edge
+                    // and the edge of the state just after the current leg index pair is a TransitBoardAlight edge,
+                    // then make that index be the final state of the current leg. This is to ensure that alerts
+                    // associated with TransitBoardAlight vertices are able to be found for legs.
+                    if (
+                        states[legIndexPairs[0]].getBackMode() != null &&
+                        states[legIndexPairs[0]].backEdge instanceof TransitBoardAlight &&
```

Comment:
Is there not some common code in this conditional that can be factored out into a shared method used by `getLastOnboardTransitState`?

Reply:
No, not here. The `getLastOnboardTransitState` operates on data that is calculated after this code is executed and splits up the states.